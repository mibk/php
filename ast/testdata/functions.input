<?php

    function a
   (  array  $a  =  0xface
   ,  int&   $b  =  'foo  b'
   ,   ...   $c
   )         {}

	function b (bool & ... $x , & ... $y,
	        string ... $z   , &     $ž  =
	           null,
		) :void{/*  */}

function get  ( ?   \
         DateTime  $d
                   ,
         )         {
         }