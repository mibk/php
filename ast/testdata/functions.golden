<?php

function a(array $a = 0xface, int &$b = 'foo  b', ...$c = ['x']) {
}

function b(bool &...$x, &...$y, string ...$z, &$ž = null): void {
}

function get(?\DateTime $d) {
}

function set($v = [0, '1', true]) {
}
